pkg/controller/vsphere/reconciler.go

```diff
@@ -38,6 +41,26 @@ const (
 	GuestInfoIgnitionEncoding = "guestinfo.ignition.config.data.encoding"
 )
 
+const (
+	machineCreationSucceeded = "MachineCreationSucceeded"
+	machineCreationFailed    = "MachineCreationFailed"
+)
+
+var (
+	conditionSuccess = vspherev1.VSphereMachineProviderCondition{
+		Type:    vspherev1.MachineCreation,
+		Status:  corev1.ConditionTrue,
+		Reason:  machineCreationSucceeded,
+		Message: "Machine successfully created",
+	}
+
+	conditionFailed = vspherev1.VSphereMachineProviderCondition{
+		Type:   vspherev1.MachineCreation,
+		Status: corev1.ConditionTrue,
+		Reason: machineCreationFailed,
+	}
+)
+
```

Comment:
Not a fan of these being package level variables, what about making functions that return new instances of the conditions so that we get a new copy each time rather than modifying the package level var?

Reply:
👍